specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/ManagedLedgerDigestUploads.json

```diff
@@ -0,0 +1,335 @@
+  "info": {
+    "version": "2022-08-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/databases/{databaseName}/ledgerDigestUploads": {
+      "get": {
+        "tags": [
+          "ManagedLedgerDigestUploads"
+        ],
+        "description": "Gets all ledger digest upload settings on a database.",
+        "operationId": "ManagedLedgerDigestUploads_ListByDatabase",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedInstanceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved ledger upload settings.",
+            "schema": {
+              "$ref": "#/definitions/ManagedLedgerDigestUploadsListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 LedgerInvalidDigestStorageTargetUpdateRequest - The database ledger digest storage upload request does not exist or has no properties object.\n\n * 400 InvalidDigestStorageEndpoint - The specified digestStorageEndpoint is invalid. It must be an Azure blob storage or Azure Confidential Ledger endpoint.\n\n * 400 MalformedDigestStorageEndpoint - The specified digest storage endpoint is a malformed URI.\n\n * 400 InvalidLedgerDigestUploadsName - The ledger digest upload name is not supported. Please use the key name “current”.\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets list of ledger digest upload settings on a database.": {
+            "$ref": "./examples/ManagedLedgerDigestUploadsList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/databases/{databaseName}/ledgerDigestUploads/{ledgerDigestUploads}": {
```

Comment:
>ledgerDigestUploads [](http://example.com/codeflow?start=181&length=19)

Name could be {ledgerDigestUpload}, as it should be singular.

Reply:
@visingla-ms : Can you please confirm if only the text in curly braces should be singular?
 
We have similar API for Azure SQL already published. https://learn.microsoft.com/en-us/rest/api/sql/2021-02-01-preview/ledger-digest-uploads/create-or-update?tabs=HTTP#ledgerdigestuploadsname

In order to maintain the customer experience for MI, we have to keep it as ledgerDigestUploads
This is PR for swagger of our existing API https://github.com/Azure/azure-rest-api-specs/pull/13871/files

Reply:
Also as per the guidelines, we have to use the plural names. reference: https://microsoft.sharepoint.com/teams/SQLServer/clouddb/_layouts/15/Doc.aspx?sourcedoc=%7Bf8fe39bf-eb2f-4975-916a-8c58abcb538d%7D&action=edit&wd=target(REST%20APIs%20and%20Clients.one%7C74c69f74-39c5-4477-a7d4-948eb72a8567%2FEnable%20your%20controller%20for%20Swagger%7C2e02e5d5-a9cd-471c-b094-9925442ab1b5%2F)&wdorigin=NavigationUrl&share=IgG_Of74L-t1SZFqjFiry1ONAUt6uaAtYgT7pwbtcKpqr60

Reply:
Yes, only the text in curly braces should be singular, even from your link "databases/{databaseName}/ " the first part is plural and second part should be singular.

Reply:
Thanks for explaining. I understand the comment now. This API is copy existing API for SQL DB. Since that API is already out with {ledgerDigestUploads} can we please get an exception for this API as well. 
Here is the link for the Swagger PR for our existing API. https://github.com/Azure/azure-rest-api-specs/pull/13871/files